### ISSUE TITLE
Regression with after 0.12.1 

### ISSUE THREAD

**Issue by nutterbg** [amidaware/tacticalrmm#1041]
**Server Info :**
 - OS: Ubuntu 20.04.4 LTS
 - Browser: Any
 - RMM Version (as shown in top left of web UI): 0.12.1

**Installation Method:**
  - [x] Standard
  - [ ] Docker

**Agent Info (please complete the following information):**
- Agent version (as shown in the 'Summary' tab of the agent from web UI): v2.0.1
- Agent OS: Windows 10 Pro, 64 bit v20H2 (build 19042.1586)

**Describe the bug**
I'm not sure if it's immediately following the upgrade, or if this regression appeared at some point after, but it seems I cannot currently change Tasks for any machines. I'm unable to remove a task that's explicitly assigned to a specific workstation. I am able to assign this task to an automation policy, but it does not apply to any of the machines under that policy. I can disable or enable a task, and it will respectively run/not run.

**(Scenario 1)**
1. Go to 'Client' -> Workstations -> any workstation (this affects the whole system)
2. Click on the 'Tasks' tab
3. Right-click the task to Delete, click OK to confirm
4. Receive "<task name> will be deleted shortly" notification
5. Click on "Refresh" button next to "Add Task"

**Expected behavior**
After Step 5 Task should no longer be visible.

**(Scenario 2)**
1. Click on Settings -> Automation Manager
2. Click on a Policy
3. Click on "Tasks" tab
4. Add a new task referencing an existing custom script (omitted detailed steps here, but can provide if needed)
5. After finishing the wizard, task appears in task list
6. After several days the task is still not applied to machines that have this automation policy in place

**Expected behavior**
After Step 6 the new task should be applied to the respective machines under that automation policy.

**Comment by nutterbg** [comment]
Just to add to this, my environment is custom as follows (though that has not been a problem so far):

Proxies ❌
Firewalls ❌
GeoIP filters ✅
fail2ban filters ❌
alternate methods of SSL cert management ✅- managed by Virtualmin
IDSs ❌
IPSs ❌
SDNs ❌
Did anything other than follow the installation instructions exactly ✅ - Tactical's Nginx is listening on additional HTTP ports for rmm, grpc and api, and is reverse-proxied behind Apache (as part of Virtualmin). Headers are passed through, and no regressions observed. NATS is exposed directly via port forward, as expected.

Thanks.

**Comment by silversword411** [comment]
You've right-clicked the policy, and told it to refresh right?

**Comment by wh1te909** [comment]
Policies are being reworked from scratch so just hold off until next release, will probably fix whatever bug you have

**Comment by nutterbg** [comment]
> You've right-clicked the policy, and told it to refresh right?

Yup, but to no avail. When I hit "See task status" next to the specific task below the policy in Automation manager, it shows that "There are no agents in this policy".

I'll wait till the next release and we'll see. Hopefully that fixes it.

I'm curious if there's a way I can debug this - perhaps a log I can look at for the service that queues up all of these actions (not sure which one that would be, NATS?)? Perhaps there's some generic troubleshooting step I can take (e.g. clear some table in the db) that would clear whatever pipe is clogged?

Thanks.

**Comment by silversword411** [comment]
Policy is getting major rework right now because of issues. I'd just hold on, and re-assess with the next release

**Comment by nutterbg** [comment]
Hi @silversword411 and @wh1te909 
I updated the problematic instance to 0.12.2 and the problem seems to have subsided.

Thanks for your time!
